db/compaction/compaction_picker_level.cc

```diff
@@ -274,21 +287,30 @@ void LevelCompactionBuilder::SetupInitialFiles() {
     }
   }
 
-  PickFileToCompact(vstorage_->ExpiredTtlFiles(), true);
+  PickFileToCompact(vstorage_->ExpiredTtlFiles(),
+                    CompactToNextLevel::kSkipLastLevel);
   if (!start_level_inputs_.empty()) {
     compaction_reason_ = CompactionReason::kTtl;
     return;
   }
 
   // Periodic Compaction
-  PickFileToCompact(vstorage_->FilesMarkedForPeriodicCompaction(), false);
+  bool periodic_compact_to_next_level =
+      ioptions_.level_compaction_dynamic_level_bytes &&
+      (mutable_cf_options_.ttl == 0 /* ttl was disabled */ ||
+       mutable_cf_options_.ttl >
```

Comment:
Should it be `ttl < periodic_compaction_seconds`?

Reply:
@cbi42 from the [comment](https://github.com/facebook/rocksdb/issues/12165#issuecomment-1866935306), we don't expect to affect the users if configured ttl <= periodic_compaction_seconds, so check if ttl > periodic_compaction_seconds.

My previous comment line above the `periodic_compaction_seconds` was wrong, ttl <= periodic_compaction_seconds will still compact to the same level and it's also fixed now. Could you please take a look again while you're free?

Reply:
By the comment, I meant that if users configure `ttl < periodic_compaction_seconds`, then the behavior change here won't matter. This is because ttl compaction will happen sooner than periodic compactions, so these files will always be compacted to the next level. So it's safe to make the behavior change to compact to next level when `ttl < periodic_compaction_seconds`. 

Now I think about it, I think doing it when `ttl > periodic_compaction_seconds` makes sense too. Maybe we can just skip the comparison with `ttl` and always compact to next level when level_compaction_dynamic_level_bytes is true. Sorry for the confusion.

Reply:
@cbi42 Thanks for your detailed explanation. I will remove this check condition.